dev/gen-protos

```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
 rm -rf pkg/proto/**/*.pb.go pkg/proto/**/*.pb.gw.go pkg/proto/**/*.swagger.json
-if ! go tool -modfile=tools/go.mod buf generate https://github.com/xmtp/proto.git#subdir=proto,branch=main; then
+if ! go tool -modfile=tools/go.mod buf generate https://github.com/xmtp/proto.git#subdir=proto,branch=nm/serialize-payer-envelope; then
```

Comment:
I do the same thing :) Develop XMTPD while I wait for the protos change to settle. 

Just making a comment here that it will have to be changed back once it's merged.